src/sql/parts/editData/execution/editDataActions.ts

```diff
@@ -87,6 +102,53 @@ export class RefreshTableAction extends EditDataAction {
 		}
 		return TPromise.as(null);
 	}
+
+	private IsQueryStringValid(queryString: string, tableName: string): boolean {
```

Comment:
What sort of queries are valid here? Are the ones we detect the only invalid cases, or are there others? For example the logic doesn't seem to have problems with multiple batches, using `GO n`, or joins. Will those all work?

Reply:
Enough edge cases keep popping up that I might want to rethink how invalid cases like this need to be handled. Maybe some sort of query metdata-based approach in SqlToolsService, rather than this fragile string parsing solution I have now.